doc/neps/nep-0029-deprecation_policy.rst

```diff
@@ -0,0 +1,190 @@
+==========================================================================================
+NEP 29 — A standard community policy for dropping support of old Python and NumPy versions
+==========================================================================================
+
+
+:Author: Thomas A Caswell <tcaswell@gmail.com>, Andreas Mueller, Brian Granger, Madicken Munk, Ralf Gommers, Matt Haberland <mhaberla@calpoly.edu>, Matthias Bussonnier <bussonniermatthias@gmail.com>, Stefan van der Walt
+:Status: Draft
+:Type: Informational Track
+:Created: 2019-07-13
+
+
+Abstract
+--------
+
+All projects across the ecosystem should adopt a common time window
+based policy for increasing the minimum version of Python and NumPy
+that downstream projects support.  By standardizing this policy
+across the community we will make it easier for downstream projects to
+plan.
+
+This is an unusual NEP in that it is recommendations for community-wide
+policy and not for changes to NumPy itself.  However, we do not
+currently have a place for SPEEPs (Scientific Python Ecosystem
+Enhancement Proposals) and given NumPy's central role in our community
+this seems like the proper place to document this.
+
+
+This is being put forward by maintainers of Matplotlib, scikit-learn,
+IPython, Jupyter, yt, SciPy, NumPy, and scikit-image.
+
+
+
+Detailed description
+--------------------
+
+When making a new major or minor release, projects should support all
+minor versions of ``Python`` initially released in the prior 42
+months from their anticipated release date and all minor version of
+NumPy released in the prior 24 months.
+
+
+The diagram::
+
+       Jan 16      Jan 17      Jan 18      Jan 19      Jan 20
+       |           |           |           |           |
+  +++++|+++++++++++|+++++++++++|+++++++++++|+++++++++++|++++++++++++
+   |              |                  |               |
+   py 3.5.0       py 3.6.0           py 3.7.0        py 3.8.0
+  |-----------------------------------------> Feb19
+            |-----------------------------------------> Dec19
+                      |-----------------------------------------> Nov20
+
+shows the support windows for ``Python``.  A project with a major or
+minor release in Feb19 should support py35 and newer, a project with a
+major or minor release in Dec19 should support py36 and newer, and a
+project with a major or minor release in Nov20 should support py37 and
+newer.
+
+The current Python release cadence is 18 months so a 42 month window
+ensures that there will always be at least two versions of ``Python``
+in the window.  By padding the window by 6 months from the anticipated
+``Python`` cadence we avoid the edge cases where a project releases
+the month after ``Python`` and would effectively only support one
```

Comment:
Why not just "last three minor Python releases", where the usual version numbering is `major:minor:micro`

Reply:
Note that there's a terminology conflict here between CPython and basically everywhere else: 3.6, 3,7, etc are frequently referred to as major releases, and 3.x.y are minor releases. So if you say "minor Python releases", people *are* going to get confused.

One way to get around that would be to refer to the Python branch status, rather than the release numbers themselves: https://devguide.python.org/#status-of-python-branches

Reply:
(Note: referring to Python feature releases as major releases is technically incorrect, as per https://devguide.python.org/devcycle/#development-cycle, so this PEP is technically correct in calling CPython's feature releases minor releases, and its bugfix releases patch releases. But there's a difference between the terminology the docs say folks *should* be using, and what they actually use)

Reply:
There is now a key at the top and given that as @charris says, the `major.minor.micro` terminology is common in the scipy community, I am inclined to leave the wording as it is.